meter/validation.cc

```diff
@@ -145,9 +145,15 @@ ValidationIssues AnalyzeTags(const Tags& tags) noexcept {
                       v_ref.get(), k_ref.get(), validated_val.get())));
     }
 
-    if (empty_or_null(k_ref) || empty_or_null(v_ref)) {
-      result.emplace(
-          ValidationIssue::Err("Tag keys or values cannot be empty"));
+    if (empty_or_null(k_ref) && empty_or_null(v_ref)) {
+      result.emplace(ValidationIssue::Err(
+          "Tags cannot be empty. Found empty key and value."));
+    } else if (empty_or_null(k_ref)) {
+      result.emplace(ValidationIssue::Err(
+          fmt::format("Empty key found with value '{}'", v_ref.get())));
```

Comment:
I would think the metric name or maybe the string representation of the id would be more useful. I'm not sure how easy it would be to map the value back to the source. At least with the metric name it is more likely I could find what was updating it in the code.

Reply:
The idea is that you're validating a given set of Tags, so the code that shows this to the user will include the string representation of the tags/id in addition to the list of validation issues.